src/HexagonOptimize.cpp

```diff
@@ -1605,6 +1616,293 @@ class OptimizeShuffles : public IRMutator {
 public:
     OptimizeShuffles(int lut_alignment) : lut_alignment(lut_alignment) {}
 };
+
+
+class VtmpyGenerator : public IRMutator {
+private:
+    using IRMutator::visit;
+
+    std::unordered_map<string, Expr> lets;
```

Comment:
Any reason you are not using Scope<Expr>? See usage in SlidingWindow.cpp or StorageFolding.cpp